runner/src/main.rs

```diff
@@ -348,6 +349,55 @@ fn build_server(opt: &BuildServer) -> Step {
     }
 }
 
+fn build_functions_server(opt: &BuildFunctionsServer) -> Step {
+    Step::Multiple {
+        name: "server".to_string(),
+        steps: vec![
+            vec![Step::Single {
+                name: "create bin folder".to_string(),
+                command: Cmd::new(
+                    "mkdir",
+                    vec!["-p".to_string(), "oak_functions/loader/bin".to_string()],
+                ),
+            }],
+            vec![Step::Single {
+                name: "build functions server".to_string(),
+                command: Cmd::new_with_env(
+                    "cargo",
+                    spread![
+                        ...match &opt.server_rust_toolchain {
+                            // This overrides the toolchain used by `rustup` to invoke the actual
+                            // `cargo` binary.
+                            // See https://github.com/rust-lang/rustup#toolchain-override-shorthand
+                            Some(server_rust_toolchain) => vec![format!("+{}", server_rust_toolchain)],
+                            None => vec![],
+                        },
```

Comment:
Could this cmd be shared with the one for building the other server? perhaps in a method parametrized with the cargo manifest file and the features to enable?

Reply:
I think this can be done easily if I use `ServerVariants` for `build_functions_server` too, instead of using `FunctionsServerVariant`. I am currently using `FunctionsServerVariant` because not all `ServerVariants` are relevant for oak-functions loader. What do you think about using `ServerVariants` for `build_functions_server` too?

Reply:
I think the structure as you implemented it makes more sense than using the variant. I was just talking about sharing part of this logic so that if we add more flags to build rust binaries in general, they will apply to both.

Reply:
I tried to refactor this into a function.